conan/tools/cmake/toolchain.py

```diff
@@ -426,32 +483,46 @@ def context(self):
             find_package_prefer_config = "OFF"
 
         os_ = self._conanfile.settings.get_safe("os")
-        android_prefix = "${CMAKE_CURRENT_LIST_DIR}" if os_ == "Android" else None
+        is_apple_ = os_ and os_ in ["Macos", "iOS", "watchOS", "tvOS"]
 
+        # Read information from host context
         host_req = self._conanfile.dependencies.host.values()
-        cross_ios = os_ in ('iOS', "watchOS", "tvOS")
-
-        # Read the buildirs
-        build_paths = []
+        host_build_paths_root = []
+        host_build_paths_noroot = []
+        host_lib_paths = []
+        host_framework_paths = []
+        host_include_paths = []
         for req in host_req:
             cppinfo = req.cpp_info.aggregated_components()
-            build_paths.extend([os.path.join(req.package_folder,
-                                       p.replace('\\', '/').replace('$', '\\$').replace('"', '\\"'))
-                                for p in cppinfo.builddirs])
-
-        if self._toolchain.find_builddirs:
-            build_paths = " ".join(['"{}"'.format(b.replace('\\', '/')
-                                                   .replace('$', '\\$')
-                                                   .replace('"', '\\"')) for b in build_paths])
-        else:
-            build_paths = ""
-
-        return {"find_package_prefer_config": find_package_prefer_config,
-                "cmake_prefix_path": "${CMAKE_CURRENT_LIST_DIR} " + build_paths,
-                "cmake_module_path": "${CMAKE_CURRENT_LIST_DIR} " + build_paths,
-                "android_prefix_path": android_prefix,
-                "cross_ios": cross_ios,
-                "generators_folder": "${CMAKE_CURRENT_LIST_DIR}"}
+            host_build_paths_root.extend([os.path.join(req.package_folder, p) for p in cppinfo.builddirs if p == ""])
```

Comment:
I think I agree we should stop adding "." as a default builddir and builddir should have effect only in `CMAKE_MODULE_PATH`, not in `CMAKE_PREFIX_PATH`. And I get why you do this, mostly because we cannot break now the model removing the "." default. But even though all this sounds complicated to understand and maintain. I'm wondering if we might postpone these fixes to 2.0 and introduce some breaking changes to the model to improve it.